ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/router-bgp.j2

```diff
@@ -34,336 +34,340 @@ router bgp {{ router_bgp.as }}
-{%         if router_bgp.aggregate_addresses[aggregate_address].as_set is arista.avd.defined(true) %}
+{%     for aggregate_address in router_bgp.aggregate_addresses | arista.avd.convert_dicts('prefix') | arista.avd.natural_sort('prefix') %}
+{%         set aggregate_address_cli = "aggregate-address " ~ aggregate_address.prefix %}
+{%         if aggregate_address.as_set is arista.avd.defined(true) %}
 {%             set aggregate_address_cli = aggregate_address_cli ~ " as-set" %}
 {%         endif %}
-{%         if router_bgp.aggregate_addresses[aggregate_address].summary_only is arista.avd.defined(true) %}
+{%         if aggregate_address.summary_only is arista.avd.defined(true) %}
 {%             set aggregate_address_cli = aggregate_address_cli ~ " summary-only" %}
 {%         endif %}
-{%         if router_bgp.aggregate_addresses[aggregate_address].attribute_map is arista.avd.defined %}
-{%             set aggregate_address_cli = aggregate_address_cli ~  " attribute-map " ~ router_bgp.aggregate_addresses[aggregate_address].attribute_map %}
+{%         if aggregate_address.attribute_map is arista.avd.defined %}
+{%             set aggregate_address_cli = aggregate_address_cli ~  " attribute-map " ~ aggregate_address.attribute_map %}
 {%         endif %}
-{%         if router_bgp.aggregate_addresses[aggregate_address].match_map is arista.avd.defined %}
-{%             set aggregate_address_cli = aggregate_address_cli ~ " match-map " ~ router_bgp.aggregate_addresses[aggregate_address].match_map %}
+{%         if aggregate_address.match_map is arista.avd.defined %}
+{%             set aggregate_address_cli = aggregate_address_cli ~ " match-map " ~ aggregate_address.match_map %}
 {%         endif %}
-{%         if router_bgp.aggregate_addresses[aggregate_address].advertise_only is arista.avd.defined(true) %}
+{%         if aggregate_address.advertise_only is arista.avd.defined(true) %}
 {%             set aggregate_address_cli = aggregate_address_cli ~ " advertise-only" %}
 {%         endif %}
    {{ aggregate_address_cli }}
 {%     endfor %}
-{%     for redistribute_route in router_bgp.redistribute_routes | arista.avd.natural_sort %}
-{%         set redistribute_route_cli = "redistribute " ~ redistribute_route %}
-{%         if router_bgp.redistribute_routes[redistribute_route].route_map is arista.avd.defined %}
-{%             set redistribute_route_cli = redistribute_route_cli ~ " route-map " ~ router_bgp.redistribute_routes[redistribute_route].route_map %}
+{%     for redistribute_route in router_bgp.redistribute_routes | arista.avd.convert_dicts('source_protocol') | arista.avd.natural_sort('source_protocol') %}
+{%         if redistribute_route.source_protocol is arista.avd.defined %}
+{%             set redistribute_route_cli = "redistribute " ~ redistribute_route.source_protocol %}
+{%         else %}
+{%             set redistribute_route_cli = "redistribute " ~ redistribute_route %}
```

Comment:
I don't think the else is required. Also, redistribute_route is now a dictionary, which can not be appended to a string.